setup.py

```diff
@@ -778,14 +758,9 @@ def run(self):
             sys.exit(1)
         
         info("Testing pyzmq-%s with libzmq-%s" % (zmq.pyzmq_version(), zmq.zmq_version()))
-        
-        try:
-            import nose
-        except ImportError:
-            warn("nose unavailable, falling back on unittest. Skipped tests will appear as ERRORs.")
-            return self.run_unittest()
-        else:
-            return self.run_nose()
+        p = Popen([sys.executable, '-m', 'pytest', '-v', os.path.join('zmq', 'tests')])
+        p.wait()
+        sys.exit(p.returncode)
```

Comment:
If we migrate this `setup.py` from `distutils` to `setuptools`, we can remove the `try-execpt` block in this patch and just add `test_requires` option to `setup` function.